include/xsimd/math/xsimd_scalar.hpp

```diff
@@ -0,0 +1,292 @@
+    using std::erfc;
+    using std::exp2;
+    using std::exp;
+    using std::expm1;
+    using std::fdim;
+    using std::fmax;
+    using std::fmin;
+    using std::floor;
+    using std::fmod;
+    using std::hypot;
+    using std::isfinite;
+    using std::isinf;
+    using std::lgamma;
+    using std::ldexp;
+    using std::log10;
+    using std::log1p;
+    using std::log2;
+    using std::log;
+    using std::nearbyint;
+    using std::proj;
+    using std::remainder;
+    using std::rint;
+    using std::rint;
+    using std::round;
+    using std::round;
+    using std::sin;
+    using std::sinh;
+    using std::sqrt;
+    using std::tan;
+    using std::tanh;
+    using std::tgamma;
+    using std::trunc;
+    using std::trunc;
+
+#ifdef XSIMD_ENABLE_XTL_COMPLEX
+    using xtl::abs;
+    using xtl::norm;
+    using xtl::proj;
+    using xtl::exp;
+    using xtl::log;
+    using xtl::log10;
+    using xtl::pow;
+    using xtl::sqrt;
+    using xtl::sin;
+    using xtl::cos;
+    using xtl::tan;
+    using xtl::asin;
+    using xtl::acos;
+    using xtl::atan;
+    using xtl::sinh;
+    using xtl::cosh;
+    using xtl::tanh;
+    using xtl::asinh;
+    using xtl::acosh;
+    using xtl::atanh;
+#endif
+
+    template <class T>
+    inline bool is_flint(const T& x)
+    {
+        return std::isnan(x - x) ? std::numeric_limits<T>::NaN() : x - std::trunc(x);
+    }
+
+    template <class T>
+    inline bool is_odd(const T& x)
+    {
+        return is_even(x - 1.);
+    }
+
+    template <class T>
+    inline bool is_even(const T& x)
+    {
+        return is_flint(x * 0.5);
+    }
+
+    template <class T>
+    inline T exp10(const T& x)
+    {
+        // FIXME: faster alternatives exist
+        return std::pow(T(10), x);
+    }
+
+    template <class T>
+    inline std::complex<T> expm1(const std::complex<T>& val)
+    {
+        // FIXME: probably not accurate
+        return std::exp(val) - 1;
+    }
+
+    template <class T0, class T1>
+    inline typename std::enable_if<std::is_integral<T1>::value, T0>::type
+    pow(const T0& t0, const T1& t1)
+    {
+      return detail::ipow(t0, t1);
+    }
+
+    template <class T0, class T1>
+    inline typename std::enable_if<std::is_floating_point<T1>::value, T0>::type
+    pow(const T0& t0, const T1& t1)
+    {
+      return std::pow(t0, t1);
+    }
+
+    template <class T0, class T1>
+    inline typename std::enable_if<std::is_integral<T1>::value, std::complex<T0>>::type
+    pow(const std::complex<T0>& t0, const T1& t1)
+    {
+      return detail::ipow(t0, t1);
+    }
+
+    template <class T0, class T1>
+    inline typename std::enable_if<!std::is_integral<T1>::value, std::complex<T0>>::type
+    pow(const std::complex<T0>& t0, const T1& t1)
+    {
+      return std::pow(t0, t1);
+    }
+
+    template <class T0, class T1>
+    inline auto
+    pow(const T0& t0, const std::complex<T1>& t1)
+    -> typename std::enable_if<std::is_scalar<T0>::value, decltype(std::pow(t0, t1))>::type
```

Comment:
one could use `std::enable_if_t` + indentation of `->`

Reply:
according to https://en.cppreference.com/w/cpp/types/enable_if ``enable_if_t`` is C++14 only, while xsimd is still C++11, if I'm not mistaken.